test/extended/pods/oom_killer.go

```diff
@@ -0,0 +1,86 @@
+package pods
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo"
+	"github.com/openshift/library-go/pkg/build/naming"
+	exutil "github.com/openshift/origin/test/extended/util"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+)
+
+var containerName = "stressng"
+
+var _ = g.Describe("[sig-node]", func() {
+	defer g.GinkgoRecover()
+
+	f := framework.NewDefaultFramework("no-oom-kills")
+	oc := exutil.NewCLIWithFramework(f)
+
+	// The goal of this test to verify if the OOM inducing pod is the one that is getting OOM Killed
+	g.It("should oom kill the pod which induced it", func() {
+		g.By("creating the OOM inducing pod")
+		podClient := f.PodClient()
+		ns := f.Namespace.Name
+		podName := naming.GetPodName("oom-inducing-pod", string(uuid.NewUUID()))
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      podName,
+				Namespace: ns,
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{{
+					Name:  containerName,
+					Image: "alexeiled/stress-ng",
+					Args:  []string{"--vm", "2", "--vm-bytes", " 10G", "--timeout", "100m"},
+				},
+				},
+			},
+		}
+		pod = podClient.Create(pod)
+		err := e2epod.WaitForPodNameRunningInNamespace(oc.KubeFramework().ClientSet, pod.Name, oc.Namespace())
+		framework.ExpectNoError(err)
+		defer func() {
+			g.By("deleting the pod")
+			podClient.Delete(context.TODO(), pod.Name, *metav1.NewDeleteOptions(0))
+		}()
+		// Collect the events to see if we're getting OOM Killed.
+		kubeClient := oc.AdminKubeClient()
+
+		events, err := kubeClient.CoreV1().Events("").List(context.TODO(), metav1.ListOptions{})
```

Comment:
Did we prove that we are seeing all the oom events?

Reply:
i kind of want to see both of those happen at the same time in this test - one critical, one not, because we'll keep adding scenarios (injected failure other pods).

I'd also like to see the node logs used to verify the oom as well as the events.

Reply:
Ok, so you want this PR and https://github.com/openshift/origin/pull/25927 to be combined? I can do that.

Reply:
Updated this to collect logs from node as well.